### ISSUE TITLE
Missing error when returning None from async function

### ISSUE THREAD

**Issue by yoav-orca** [python/mypy#9191]
Note: if you are reporting a wrong signature of a function or a class in
the standard library, then the typeshed tracker is better suited
for this report: https://github.com/python/typeshed/issues

Please provide more information to help us understand the issue:

* Are you reporting a bug, or opening a feature request?
I'm reporting a Bug

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.
```python
def ret_nothing() -> None:
    pass


def use_nothing() -> None:
    ret = ret_nothing()


async def async_ret_nothing() -> None:
    pass


async def async_use_nothing() -> None:
    ret = await async_ret_nothing()

```

* What is the actual behavior/output?
```
mypy --strict test.py
test.py:6: error: "ret_nothing" does not return a value
Found 1 error in 1 file (checked 1 source file)
```

* What is the behavior/output you expect?
I expected mypy to report: `error: "async_ret_nothing" does not return a value`

* What are the versions of mypy and Python you are using?
python 3.8.4, mypy 0.782
  Do you see the same issue after installing mypy from Git master?
Yes, tested with mypy 0.790+dev.a5455bd9f37825d24415e46a4d278d71502c9d07

* What are the mypy flags you are using? (For example --strict-optional)
`--strict`

**Comment by ilevkivskyi** [comment]
This is likely because detection of `None` return happens immediately when checking a function call, we probably need to special-case await expression (async functions have return type auto-wrapped in `Awaitable[...]`). This is low priority however IMO.

**Comment by heejaechang** [comment]
another issue to take a look. for this

```python
async def asyncGen():
    await asyncio.sleep(1)
    yield 1
```

we show return type as `Generator` rather than `AsyncGenerator` but when `asyncGen` is used in `async for ...`, it shows its return type as `AsyncGenerator`

**Comment by hauntsaninja** [comment]
The original issue is fixed. I'm not sure what your issue is, you'll need to provide more of an example. Your function is unannotated, so mypy will just treat it as `def () -> Any`

**Comment by heejaechang** [comment]
@hauntsaninja oops, sorry, I left comment on wrong issue.
